pkg/cmd/cli/cmd/process.go

```diff
@@ -172,19 +173,14 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
 		return nil
 	}
 
-	label := kcmdutil.GetFlagString(cmd, "labels")
-	if len(label) != 0 {
-		lbl := ctl.ParseLabels(label)
-		for key, value := range lbl {
-			templateObj.ObjectLabels[key] = value
-		}
-	}
-
 	result, err := client.TemplateConfigs(namespace).Create(templateObj)
 	if err != nil {
 		return err
 	}
 
+	label := kcmdutil.GetFlagString(cmd, "labels")
```

Comment:
@kargakis you can have some label set in the template that will contain parameter. By overriding it, you basically changing the template before it gets chance to process. I think we should be only adding the labels to the result Config/List as the object you are going to create. You are not creating Template, just processing it.